packages/protocol/src/action/request/Specifier.ts

```diff
@@ -144,3 +164,62 @@ export namespace Specifier {
         return request.endpoint?.number;
     }
 }
+
+export function toWildcardOrHexPath(name: string, value: number | bigint | undefined) {
+    if (value === undefined) {
+        return "*";
+    }
+    return `${name ? `${name}:` : ""}0x${value.toString(16)}`;
+}
+
+/**
+ * Resolve a path into a human readable textual form for logging
+ * TODO: Add a Diagnostic display formatter for this
```

Comment:
I added diagnostic display to `DataModelPath` to handle formatting of paths in general, and `ExpandedPath` to handle conversion of wire paths to `DataModelPath`.

You might add the "!" and "[ADD]" flags there or handle them separately, but otherwise would be good if we standardized on `ExpandedPath`.